src/anomalib/callbacks/metrics.py

```diff
@@ -182,16 +181,37 @@ def _update_metrics(
         image_metric.update(output["pred_scores"], output["label"].int())
         if "mask" in output and "anomaly_maps" in output:
             pixel_metric.to(self.device)
-            pixel_metric.update(torch.squeeze(output["anomaly_maps"]), torch.squeeze(output["mask"].int()))
+            if "semantic_mask" in output:
+                self._update_pixel_metrics(pixel_metric, output)
+            else:
+                pixel_metric.update(torch.squeeze(output["anomaly_maps"]), torch.squeeze(output["mask"].int()))
```

Comment:
I'm not happy with this part of the design, as I feel that the metric callback should not contain logic related to specific metrics.

Reply:
agree, this part should be more generic and does not contain any specific implementation.

Reply:
It also goes against the intended use of `MetricCollection`, which is to group metrics together that operate over the same target and predictions. The sPRO metric operates over semantic masks instead of binary masks, so technically should be in a different metric collection compared to the other pixel metrics.